libtenzir/builtins/functions/uuid.cpp

```diff
@@ -0,0 +1,110 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/    Visibility
+//   | |/ / __ |_\ \  / /        Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2025 The Tenzir Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/arrow_utils.hpp>
+#include <tenzir/concept/parseable/tenzir/si.hpp>
+#include <tenzir/detail/narrow.hpp>
+#include <tenzir/fbs/aggregation.hpp>
+#include <tenzir/flatbuffer.hpp>
+#include <tenzir/series_builder.hpp>
+#include <tenzir/tql2/eval.hpp>
+#include <tenzir/tql2/plugin.hpp>
+#include <tenzir/uuid.hpp>
+
+#include <arrow/builder.h>
+#include <boost/uuid/uuid.hpp>
+#include <boost/uuid/uuid_generators.hpp>
+#include <boost/uuid/uuid_io.hpp>
+
+#include <memory>
+#include <string>
+#include <variant>
+
+namespace tenzir::plugins::uuid {
+
+TENZIR_ENUM(uuid_version, nil, v1, v4, v6, v7);
+
+namespace {
+
+using uuid_generator_type = std::variant<boost::uuids::nil_generator,     // nil
+                                         boost::uuids::time_generator_v1, // v1
+                                         boost::uuids::random_generator,  // v4
+                                         boost::uuids::time_generator_v6, // v6
+                                         boost::uuids::time_generator_v7  // v7
+                                         >;
+
+class uuid final : public function_plugin {
+public:
+  auto name() const -> std::string override {
+    return "tql2.uuid";
+  }
+
+  auto make_function(invocation inv, session ctx) const
+    -> failure_or<function_ptr> override {
+    auto version_opt = std::optional<located<std::string>>{};
+    TRY(argument_parser2::function("uuid")
+          .named("version", version_opt)
+          .parse(inv, ctx));
+    using namespace std::string_literals;
+    auto version = "v4"s;
+    if (version_opt) {
+      version = version_opt->inner;
+    }
```

Comment:
Do we want to check for empty string?

Reply:
I don't think we care, because `from_string<T>` immediately afterwards would handle this.